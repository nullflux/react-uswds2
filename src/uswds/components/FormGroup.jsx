import React from 'react';
import PropTypes from 'prop-types';

/**
 * A form group wraps a related set of form elements
 *
 * Required props:
 * - children: node, usually text to display within the tag
 *
 * Optional props:
 * - classes: string, additional classes
 */

export default class FormGroup extends React.Component {
  
  static propTypes = {
    children: PropTypes.node.isRequired,
    hasError: PropTypes.bool
  };

  static defaultProps = {
    hasError: false
  };

  render() {
    return (
      <div className={`usa-form-group ${this.props.hasError ? 'usa-form-group--error' : ''}`}>
        {this.props.children}
      </div>
    );
  }
}