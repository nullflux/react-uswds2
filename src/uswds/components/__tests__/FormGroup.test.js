import React from 'react';
import { mount } from 'enzyme';
import FormGroup from '../FormGroup.jsx';
import TextInput from '../TextInput.jsx';

jest.unmock('../TextInput.jsx');
jest.unmock('../FormGroup.jsx');

describe('FormGroup', () => {
  it('should render correctly', () => {
    let component = null;
    const label = 'label goes here';
    const testId = 'myId';

    component = mount(
    <FormGroup>
      <TextInput id={testId} label={label} />
    </FormGroup>);
    expect(component).toMatchSnapshot();
  });
});