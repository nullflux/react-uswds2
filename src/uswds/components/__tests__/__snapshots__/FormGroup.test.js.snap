// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should render correctly 1`] = `
<FormGroup
  hasError={false}
>
  <div
    className="usa-form-group "
  >
    <TextInput
      classes=""
      enableSpellCheck={false}
      id="myId"
      label="label goes here"
      required={false}
      type="text"
    >
      <InputLabel
        hasError={false}
        htmlFor="myId"
        label="label goes here"
        required={false}
      >
        <label
          className="usa-label"
          htmlFor="myId"
          id="myId-label"
        >
          label goes here
        </label>
      </InputLabel>
      <input
        aria-labelledby="myId-label"
        aria-required={false}
        className="usa-input  null"
        id="myId"
        name="myId"
        onBlur={[Function]}
        onChange={[Function]}
        required={false}
        spellCheck={false}
        type="text"
        value=""
      />
    </TextInput>
  </div>
</FormGroup>
`;
